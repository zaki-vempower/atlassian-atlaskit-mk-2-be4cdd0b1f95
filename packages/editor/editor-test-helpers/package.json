{
  "name": "@atlaskit/editor-test-helpers",
  "version": "10.3.2",
  "private": true,
  "description": "Test Helpers for Atlassian Editor components",
  "license": "Apache-2.0",
  "atlaskit:src": "src/index.ts",
  "module": "dist/esm/index.js",
  "main": "dist/cjs/index.js",
  "types": "dist/cjs/index.d.ts",
  "author": "Atlassian Pty Ltd",
  "maintainers": [
    "Stanislav S"
  ],
  "atlaskit": {
    "internal": true,
    "name": "Editor Test Helpers",
    "team": "editor"
  },
  "repository": "https://bitbucket.org/atlassian/atlaskit-mk-2",
  "dependencies": {
    "@atlaskit/activity": "^1.0.1",
    "@atlaskit/adf-schema": "^4.3.1",
    "@atlaskit/analytics-gas-types": "^4.0.12",
    "@atlaskit/analytics-listeners": "^6.2.2",
    "@atlaskit/analytics-next": "^6.3.2",
    "@atlaskit/avatar": "^17.1.2",
    "@atlaskit/button": "^13.3.2",
    "@atlaskit/editor-common": "^43.0.0",
    "@atlaskit/editor-core": "^115.2.1",
    "@atlaskit/icon": "^19.0.11",
    "@atlaskit/media-core": "^31.0.1",
    "@atlaskit/media-test-helpers": "^25.2.4",
    "@atlaskit/smart-card": "^12.6.1",
    "@atlaskit/textarea": "^2.2.1",
    "@types/enzyme": "^3.1.15",
    "@types/react": "^16.8.0",
    "enzyme": "^3.10.0",
    "prosemirror-commands": "^1.0.7",
    "prosemirror-keymap": "^1.0.1",
    "prosemirror-model": "^1.7.0",
    "prosemirror-state": "^1.2.2",
    "prosemirror-tables": "0.9.5",
    "prosemirror-utils": "^0.9.6",
    "prosemirror-view": "^1.9.12",
    "react-intl": "^2.6.0"
  },
  "peerDependencies": {
    "react": "^16.8.0"
  },
  "devDependencies": {
    "@types/prosemirror-commands": "^1.0.1",
    "@types/prosemirror-keymap": "^1.0.1",
    "@types/prosemirror-model": "^1.7.2",
    "@types/prosemirror-state": "^1.2.0",
    "@types/prosemirror-view": "^1.9.0",
    "chai": "^4.1.2",
    "sinon": "^2.2.0"
  }
}
