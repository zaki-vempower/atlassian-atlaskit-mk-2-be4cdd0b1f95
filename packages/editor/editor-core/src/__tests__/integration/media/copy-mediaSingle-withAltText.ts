import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import { editable, getDocFromElement, fullpage } from '../_helpers';
import { testMediaFileId } from '@atlaskit/editor-test-helpers';
import {
  mountEditor,
  goToEditorTestingExample,
} from '../../__helpers/testing-example-helpers';

const baseADFWithMediaAltText = {
  version: 1,
  type: 'doc',
  content: [
    {
      type: 'mediaSingle',
      attrs: {
        width: 66.67,
        layout: 'wrap-left',
      },
      content: [
        {
          type: 'media',
          attrs: {
            id: testMediaFileId,
            type: 'file',
            collection: 'MediaServicesSample',
            width: 2378,
            height: 628,
            alt: 'test',
          },
        },
      ],
    },
    {
      type: 'paragraph',
      content: [],
    },
  ],
};

BrowserTestCase(
  'Copy paste a media single with alt text properly',
  { skip: ['edge', 'ie', 'safari'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(baseADFWithMediaAltText),
      media: {
        allowMediaSingle: true,
        UNSAFE_allowAltTextOnImages: true,
      },
    });

    await page.isVisible('.media-single');
    await page.waitForSelector('.ProseMirror :nth-child(1) .media-single');
    await page.click('.ProseMirror :nth-child(1) .media-single');
    await page.copy();

    await page.keys(['ArrowDown']);

    await page.type(editable, 'pasting');

    await page.paste();

    await page.waitForSelector('.ProseMirror :nth-child(3) .media-single');

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
