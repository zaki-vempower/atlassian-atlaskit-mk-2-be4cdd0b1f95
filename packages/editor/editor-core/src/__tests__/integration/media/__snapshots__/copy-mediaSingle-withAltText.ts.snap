// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy paste a media single with alt text properly 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "wrap-left",
        "width": 66.67,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": "DUMMY-OBJECT-ID",
            "__displayType": null,
            "__external": false,
            "__fileMimeType": "image/svg+xml",
            "__fileName": "example.png",
            "__fileSize": 235,
            "alt": "test",
            "collection": "MediaServicesSample",
            "height": 628,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 2378,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "content": Array [
        Object {
          "text": "pasting",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "layout": "wrap-left",
        "width": 66.67,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": "DUMMY-OBJECT-ID",
            "__displayType": null,
            "__external": false,
            "__fileMimeType": "image/svg+xml",
            "__fileName": "example.png",
            "__fileSize": 235,
            "alt": "test",
            "collection": "MediaServicesSample",
            "height": 628,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 2378,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
